src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -9,19 +9,38 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     internal class ProvideSemanticTokensResponse
     {
-        public ProvideSemanticTokensResponse(SemanticTokens result, long? hostDocumentSyncVersion)
+        public ProvideSemanticTokensResponse(SemanticTokens? result, SemanticTokensDelta? editResult, long? hostDocumentSyncVersion)
         {
             Result = result;
+            EditResult = editResult;
             HostDocumentSyncVersion = hostDocumentSyncVersion;
         }
 
-        public SemanticTokens Result { get; }
+        public SemanticTokens? Result { get; }
+
+        public SemanticTokensDelta? EditResult { get; }
 
         public long? HostDocumentSyncVersion { get; }
 
-        public override bool Equals(object obj) =>
-            obj is ProvideSemanticTokensResponse other &&
-            other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion) && other.Result.Equals(Result);
+        public override bool Equals(object obj)
```

Comment:
I should have left comments here, I remember that there was a VERY specific problem which required this Equals behavior but not what it was. Definitely verify by running the experimental instance through its paces and making sure nothing falls over.

Reply:
Thanks for the heads up! I ended up created a new class `ProvideSemanticTokensEditsResponse` instead of modifying the existing class. From testing everything appears to be ok.